src/commands/version.ts

```diff
@@ -1,39 +1,59 @@
-import {Command, Flags} from '@oclif/core'
+import {CliUx, Command, Flags} from '@oclif/core'
 import {EOL, type as osType, release as osRelease} from 'node:os'
+import {exec} from 'shelljs'
+
+interface VersionDetail {
+  cliVersion: string;
+  architecture: string;
+  nodeVersion: string;
+  pluginVersions?: string[];
+  osVersion?: string;
+}
 
 export default class Version extends Command {
+  static enableJsonFlag = true
+
   public static flags = {
     verbose: Flags.boolean({
       summary: 'Show additional information about the CLI.',
       description: 'Additionally shows the architecture, node version, operating system, and versions of plugins that the CLI is using.',
     }),
   }
 
-  async run(): Promise<void> {
+  async run(): Promise<VersionDetail> {
     const {flags} = await this.parse(Version)
-    let output = `${this.config.userAgent}${EOL}`
 
-    if (flags.verbose) {
-      const versions = this.config.userAgent.split(' ')
-      const cliVersion: string = versions[0]
-      const architecture: string = versions[1]
-      const nodeVersion: string = versions[2]
+    const versions = this.config.userAgent.split(' ')
+    const cliVersion = versions[0]
+    const architecture = versions[1]
+    const nodeVersion = versions[2]
 
-      let pluginVersions = ''
-      for (const plugin of this.config.plugins) {
-        pluginVersions += `${EOL}\t${plugin.name}@${plugin.version} (${plugin.type})`
-      }
+    const versionDetail:VersionDetail = {cliVersion, architecture, nodeVersion}
+    let output = `${cliVersion} ${architecture} ${nodeVersion}`
 
+    if (flags.verbose) {
+      const pluginVersion: string = exec('./bin/run plugins --core', {
+        silent: true,
+      }).toString()
+      const pluginVersions: string[] = pluginVersion.split('\n')
+      pluginVersions.pop()
       const osVersion = `${osType()} ${osRelease()}`
 
+      versionDetail.pluginVersions = pluginVersions
+      versionDetail.osVersion = osVersion
+
       output = ` CLI Version : ${EOL}\t${cliVersion}
-${EOL} Architecture: ${EOL}\t${architecture}
-${EOL} Node Version : ${EOL}\t${nodeVersion}
-${EOL} Plugin Version: ${pluginVersions}
-${EOL} OS and Version: ${EOL}\t${osVersion}
-`
+      ${EOL} Architecture: ${EOL}\t${architecture}
+      ${EOL} Node Version : ${EOL}\t${nodeVersion}
+      ${EOL} Plugin Version: ${pluginVersions.join('\n\t')}
+      ${EOL} OS and Version: ${EOL}\t${osVersion}
+      `
+    }
+
+    if (!flags.json) {
+      CliUx.ux.log(output)
```

Comment:
Why not just use [`this.log()`](https://github.com/oclif/core/blob/22ced3cb167bb291a1e18c8621264f76dd6646a9/src/command.ts#L202-L208) which already includes logic for whether or not the `--json` flag is passed in?